lib/fizzy/parser_expr.cpp

```diff
@@ -157,34 +157,27 @@ inline uint8_t get_branch_arity(const ControlFrame& frame) noexcept
     return get_branch_frame_type(frame).has_value() ? 1 : 0;
 }
 
-inline void validate_branch_stack_height(const ControlFrame& current_frame,
-    const ControlFrame& branch_frame, const Stack<ValType>& operand_stack)
+inline void update_branch_stack(const ControlFrame& current_frame, const ControlFrame& branch_frame,
+    Stack<ValType>& operand_stack)
 {
-    const auto current_frame_stack_height = static_cast<int>(operand_stack.size());
+    assert(static_cast<int>(operand_stack.size()) >= current_frame.parent_stack_height);
 
-    assert(current_frame_stack_height >= current_frame.parent_stack_height);
-
-    const auto arity = get_branch_arity(branch_frame);
-    if (!current_frame.unreachable &&
-        (current_frame_stack_height < current_frame.parent_stack_height + arity))
-        throw validation_error{"branch stack underflow"};
+    const auto branch_frame_type = get_branch_frame_type(branch_frame);
+    if (branch_frame_type.has_value())
+        drop_operand(current_frame, operand_stack, *branch_frame_type);
 }
 
-
 void push_branch_immediates(const ControlFrame& branch_frame, int stack_height, bytes& immediates)
 {
-    const auto arity = get_branch_arity(branch_frame);
-
     // How many stack items to drop when taking the branch.
-    // This value can be negative for unreachable instructions.
-    const auto stack_drop = stack_height - branch_frame.parent_stack_height - arity;
+    const auto stack_drop = stack_height - branch_frame.parent_stack_height;
 
     // Push frame start location as br immediates - these are final if frame is loop,
     // but for block/if/else these are just placeholders, to be filled at end instruction.
     push(immediates, static_cast<uint32_t>(branch_frame.code_offset));
     push(immediates, static_cast<uint32_t>(branch_frame.immediates_offset));
     push(immediates, static_cast<uint32_t>(stack_drop));
-    push(immediates, arity);
+    push(immediates, get_branch_arity(branch_frame));
```

Comment:
What is the return type of `get_branch_arity`?

Reply:
It's `uint8_t`

Reply:
I know the code prior was not clear too (with using auto), but can you leave a comment here stating it is `uint8_t`?

We have a separate PR which tries to change the types to unified `uint32_t`, but that showed slowdown: #376

Reply:
Added comment.